### PR TITLE
Aggregation fixes

### PR DESCRIPTION
**Pull request by jgoizueta**
See #889 & #1001

### CODE REVIEW

**Comment by jgoizueta** [comment]
@Algunenano can you check the tests added here (including the one that is skipped) in your cutting-edge config?

**Comment by Algunenano** [comment]
I'm getting 2 timeouts that arent' there in master:
```
  2) aggregation mvt (postgis) for default each aggr. cell is in a single tile:
     Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
  

  3) aggregation mvt (postgis) for point-sample each aggr. cell is in a single tile:
     Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

**Comment by jgoizueta** [comment]
Pending stuff:
- [ ] benchmarking: I'll setup a staging dedicated server to compare performance with production
- [ ] sort out what's going with trunk Postgis test timeouts (@Algunenano 🙏 could you take a look when you have some time?)

**Comment by Algunenano** [comment]
About the timings:
- Using:
  - Node 9
  - Postgresql 11
  - Postgis trunk
  - Geos master
  - Gdal 2.3.1
  - Proj 5.1.0
  - Installing using npm (clean environment without package-lock.json) using prebuilt binaries (no dynamic). 
  - The default test configuration. (`cp config/environments/test.js.example config/environments/test.js`)

- Dropping back to node 6 didn't make a difference.
Node 9:
```
$ ./node_modules/.bin/_mocha -c -u tdd -t 10000 test/acceptance/aggregation.js --grep "for default each aggr. cell is in a single tile"


  aggregation
    mvt (mapnik)
      ✓ for default each aggr. cell is in a single tile (4072ms)
    mvt (postgis)
      ✓ for default each aggr. cell is in a single tile (6962ms)


  2 passing (11s)
```

Node 6:
```
./node_modules/.bin/_mocha -c -u tdd -t 10000 test/acceptance/aggregation.js --grep "for default each aggr. cell is in a single tile"


  aggregation
    mvt (mapnik)
      ✓ for default each aggr. cell is in a single tile (4185ms)
    mvt (postgis)
      ✓ for default each aggr. cell is in a single tile (6826ms)
```

* Moving to the laptop with PG 11 - beta11 also didn't help:
```
$ ./node_modules/.bin/_mocha -c -u tdd -t 10000 test/acceptance/aggregation.js --grep "for default each aggr. cell is in a single tile"


  aggregation
    mvt (mapnik)
      ✓ for default each aggr. cell is in a single tile (7783ms)
    mvt (postgis)
      ✓ for default each aggr. cell is in a single tile (7182ms)

  2 passing (15s)
```

* Once that was checked I've started looking into the queries:
I started analyzing what queries are launched during the process then noticed there were quite a few. I'm attaching the logs ([aggregation.log](https://github.com/CartoDB/Windshaft-cartodb/files/2201164/aggregation.log)). There are ~40 queries to Postgres. This looks really bad.

I'm running only one single test:
```
  aggregation
    mvt (mapnik)
      ✓ for default each aggr. cell is in a single tile (4072ms)
```
And all I see is one map instantiation and 4 getTile calls.


Now checking the queries themselves, if I can see in the analyze output that there are some many functions and the calculated cost is so high that JIT is being triggered. Disabling JIT drops the time of that test from 4 seconds to 2.8, which is still 10x more than the timings in Travis.

**Comment by Algunenano** [comment]
After switching to `UNION ALL` there are no longer timeouts with PG11:
```
  1190 passing (2m)
  18 pending
  1 failing ---- (unrelated)
```

**Comment by jgoizueta** [comment]
It's weird but I'd say that here *actual* and *expected* are swapped:
```
  2) aggregation mvt (postgis) for default no partially aggregated cells:

      Uncaught AssertionError [ERR_ASSERTION]: [ { _cdb_feature_count: 3, cartodb_id: 2 } ] deepEqual [ { cartodb_id: 2, _cdb_feature_count: 2 },
  { cartodb_id: 3, _cdb_feature_count: 1 } ]
      + expected - actual

       [
         {
      -    "_cdb_feature_count": 3
      +    "_cdb_feature_count": 2
           "cartodb_id": 2
         }
      +  {
      +    "_cdb_feature_count": 1
      +    "cartodb_id": 3
      + 
```

**Comment by Algunenano** [comment]
That test is now passing (now sure what did it, parallelism? jit?) so I'd leave it for now.

**Comment by jgoizueta** [comment]
@dgaubert @enekid This seems teady to roll, can you take a look?

File: test/acceptance/aggregation.js [hunk 1]
```diff
@@ -2294,9 +2625,9 @@ describe('aggregation', function () {
 
                                 options: {
                                     sql: POINTS_SQL_PAIRS,
-                                    resolution: 1,
                                     aggregation: {
-                                        threshold: 1
+                                        threshold: 1,
```
Comment (dgaubert) [inline test/acceptance/aggregation.js]: ❤️

File: test/acceptance/aggregation.js [hunk 2]
```diff
@@ -134,6 +134,100 @@ describe('aggregation', function () {
     from generate_series(0, 13) x
     `;
 
+    // Some points at corners and centers of aggregation grids (zoom=1, resolution=1)
```
Comment (dgaubert) [inline test/acceptance/aggregation.js]: ❤️

**Pull request merged**